timemachine/cpp/src/nonbonded_interaction_group.cu

```diff
@@ -280,35 +292,43 @@ void NonbondedInteractionGroup<RealType, Interpolated>::execute_device(
     // identify which tiles contain interpolated parameters
 
     const int tpb = 32;
-    const int B = (N + tpb - 1) / tpb;
-
-    dim3 dimGrid(B, 3, 1);
+    const int B1 = ceil_divide(N1_, tpb); // number of blocks for group 1
+    const int B2 = ceil_divide(N2_, tpb); // number of blocks for group 2
 
     // (ytz) see if we need to rebuild the neighborlist.
     // (ytz + jfass): note that this logic needs to change if we use NPT later on since a resize in the box
     // can introduce new interactions.
-    k_check_rebuild_coords_and_box<RealType>
-        <<<B, tpb, 0, stream>>>(N, d_x, d_nblist_x_, d_box, d_nblist_box_, nblist_padding_, d_rebuild_nblist_);
+    k_check_rebuild_coords_and_box_gather<RealType><<<B1, tpb, 0, stream>>>(
+        N1_, d_atom_idxs_1_, d_x, d_nblist_x_, d_box, d_nblist_box_, nblist_padding_, d_rebuild_nblist_);
+
+    k_check_rebuild_coords_and_box_gather<RealType><<<B2, tpb, 0, stream>>>(
+        N2_, d_atom_idxs_2_, d_x, d_nblist_x_, d_box, d_nblist_box_, nblist_padding_, d_rebuild_nblist_);
```

Comment:
todo: should be able to do this with a single kernel call by concatenating `d_atom_idxs_*` first